packages/earth-admin/src/pages-client/users/home.tsx

```diff
@@ -0,0 +1,318 @@
+      include: 'groups',
+    };
+    return encodeQueryToURL('users', query);
+  };
+
+  const { watch, setValue, control, getValues } = useForm({
+    mode: 'onChange',
+  });
+
+  const inviteUsersWatcher = watch('users');
+
+  useEffect(() => {
+    (async () => {
+      const groupsResponse: any = await getAvailableGroups(selectedGroup);
+      const groups = groupsResponse.data.map((item) => ({ label: item.name, value: item.id }));
+      setAvailableGroups(groups);
+      setValue(
+        'role',
+        groups.find((group) => group.label.includes('VIEWER'))
+      );
+    })();
+  }, []);
+
+  const customStylesUsers = {
+    ...CUSTOM_STYLES,
+    multiValueLabel: (provided, state) => {
+      return {
+        ...provided,
+        color: state.data.hasError
+          ? 'red'
+          : state.data.hasSuccess
+          ? 'green'
+          : 'var(--marapp-gray-9)',
+        borderRadius: '50px',
+        display: 'flex',
+      };
+    },
+    multiValueRemove: (provided, state) => ({
+      ...provided,
+      color: state.data.hasError ? 'red' : 'var(--marapp-gray-9)',
+    }),
+    control: (provided, state) => {
+      return {
+        ...provided,
+        minHeight: '55px',
+      };
+    },
+  };
+
+  const customStylesRoles = {
+    ...CUSTOM_STYLES,
+    control: (provided, state) => {
+      return {
+        ...provided,
+        minHeight: '55px',
+      };
+    },
+  };
+
+  const addUsersHandler = async () => {
+    setServerErrors([]);
+    setUsersFeedback([]);
+    setIsLoadingInviteUsers(true);
+
+    const { users, role } = getValues();
+
+    try {
+      const result: any = await addUsers(
+        {
+          emails: users.map((user) => user.value),
+          groups: [role?.value],
+        },
+        selectedGroup
+      );
+
+      const feedback = [];
+
+      const feedbackUsers = result.data.map((item) => {
+        const hasError = !!item.error;
+        const hasSuccess = !!item.success;
+
+        feedback.push({
+          hasError,
+          hasSuccess,
+          title: item.email,
+          detail: item.error,
+        });
+
+        return {
+          label: item.email,
+          value: item.email,
+          hasError,
+          hasSuccess,
+        };
+      });
+
+      setValue('users', feedbackUsers);
+      setUsersFeedback(feedback);
+    } catch (err) {
+      setServerErrors(err.data?.errors || err.data);
+    }
+
+    setIsLoadingInviteUsers(false);
+  };
+
+  const cancelUsersHandler = () => {
+    setValue('users', []);
+    setServerErrors([]);
+    setUsersFeedback([]);
+  };
+
+  const { listProps: userListProps, mutate } = useInfiniteList(getQuery, getAllUsers);
+
+  return (
+    writePermissions && (
+      <ContentLayout className="marapp-qa-usershome">
+        {writePermissions && (
+          <>
+            <div className="ng-grid">
+              <div className="ng-width-1-1 ng-margin-bottom">
+                <InlineEditCard submitButtonText="Add users">
```

Comment:
```suggestion
                <Card>
```
This will never need inline editing functionality